eng/native/configurecompiler.cmake

```diff
@@ -193,6 +193,10 @@ if(CLR_CMAKE_HOST_LINUX)
   add_compile_options($<$<COMPILE_LANGUAGE:ASM>:-Wa,--noexecstack>)
   add_linker_flag(-Wl,--build-id=sha1)
   add_linker_flag(-Wl,-z,relro,-z,now)
+  if(CLR_CMAKE_HOST_ALPINE_LINUX)
+    add_linker_flag(-lucontext)
+	add_linker_flag(-Wl,-z,notext)
```

Comment:
What doesn't work without libucontext? This file is used by coreclr, libs and corehost as well, none of which every required this additional dependency.

libucontext is a separate package dependency which we should try not to introduce unless it's necessary (in which case it should be moved near to the cmake file of component needing it).

Reply:
If memory serves, libunwind was needing it on `x86`.

Reply:
I was wrong, ildasm also wants to use `setcontext` symbols, which are included natively in `glibc`
```
 [100%] Linking CXX executable ilasm
  ld.lld: error: undefined symbol: setcontext
  >>> referenced by Gos-linux.c:309 (/var/build/dotnet7/testing/dotnet7-stage0/src/dotnet-e6dd91c290b808f971a1ac69c2fb29395bbf1051/src/runtime/src/native/external/libunwind/src/x86/Gos-linux.c:309)
  >>>               Los-linux.c.o:(_ULx86_local_resume) in archive ../pal/src/libcoreclrpal.a
  clang-15: error: linker command failed with exit code 1 (use -v to see invocation)
  make[3]: *** [ilasm/CMakeFiles/ilasm.dir/build.make:267: ilasm/ilasm] Error 1
  make[2]: *** [CMakeFiles/Makefile2:2715: ilasm/CMakeFiles/ilasm.dir/all] Error 2
  make[2]: *** Waiting for unfinished jobs....
  [100%] Linking CXX executable ildasm
  ld.lld: error: undefined symbol: setcontext
  >>> referenced by Gos-linux.c:309 (/var/build/dotnet7/testing/dotnet7-stage0/src/dotnet-e6dd91c290b808f971a1ac69c2fb29395bbf1051/src/runtime/src/native/external/libunwind/src/x86/Gos-linux.c:309)
  >>>               Los-linux.c.o:(_ULx86_local_resume) in archive ../../pal/src/libcoreclrpal.a
  clang-15: error: linker command failed with exit code 1 (use -v to see invocation)
  make[3]: *** [ildasm/exe/CMakeFiles/ildasm.dir/build.make:385: ildasm/exe/ildasm] Error 1
  make[2]: *** [CMakeFiles/Makefile2:2750: ildasm/exe/CMakeFiles/ildasm.dir/all] Error 2
  make[1]: *** [CMakeFiles/Makefile2:1621: CMakeFiles/iltools.dir/rule] Error 2
  make: *** [Makefile:234: iltools] Error 2
  ~/dotnet7/testing/dotnet7-stage0/src/dotnet-e6dd91c290b808f971a1ac69c2fb29395bbf1051/src/runtime/src/coreclr
  Failed to build "CoreCLR component".
```

Reply:
Too quick to the punch, indeed still libunwind but the change in location isn't working for some reason.